librb/include/rb_memory.h

```diff
@@ -37,6 +37,7 @@
 
 void rb_outofmemory(void) __attribute__((noreturn));
 
+__attribute__((malloc, alloc_size(1), returns_nonnull))
```

Comment:
clang does not recognise the alloc_size attribute yet, can we provide an alternative set of attributes on clang?